File: lifebook/android/app/src/main/java/com/beyondpandora/hughmanatee/data/worker/WorkerModels.kt
package com.beyondpandora.hughmanatee.data.worker

import kotlinx.serialization.Serializable

/**
 * Worker API models — exact Kotlin equivalents of the TypeScript
 * interfaces in app/src/services/worker.ts.
 */

// ── /agent/config ──────────────────────────────────────────

@Serializable
data class AgentConfigRequest(
    val first_name: String,
    val birth_year: Int? = null,
    val hometown: String? = null,
    val voice_id: String,
    val last_anchor: String? = null,
    val preference: String? = null, // "old" | "recent"
)

@Serializable
data class RuntimeContext(
    val seed_prompts: List<String>,
    val era_hooks: List<String>,
)

@Serializable
data class AgentConfigResponse(
    val agent_id: String,
    val conversation_token: String,
    val first_turn: String,
    val runtime_context: RuntimeContext,
)

// ── /collage/images ────────────────────────────────────────

@Serializable
data class CollageRequest(
    val birth_year: Int? = null,
    val hometown: String? = null,
    val theme: String? = null,
)

@Serializable
data class CollageResponse(
    val images: List<CollageImage>,
    val gradient: GradientColors,
)

@Serializable
data class CollageImage(
    val url: String,
    val alt: String,
)

@Serializable
data class GradientColors(
    val from: String,
    val to: String,
)

// ── /session/anchor ────────────────────────────────────────

@Serializable
data class AnchorRequest(
    val turns: List<TurnPayload>,
)

@Serializable
data class TurnPayload(
    val speaker: String, // "user" | "hugh"
    val text: String,
)

@Serializable
data class AnchorResponse(
    val anchor_phrase: String,
    val title_suggestion: String,
    val entities: List<ExtractedEntity>,
)

@Serializable
data class ExtractedEntity(
    val kind: String, // "person" | "place" | "object" | "date" | "event"
    val value: String,
)

// ── Generic error ──────────────────────────────────────────

@Serializable
data class WorkerError(
    val error: String,
)

File: lifebook/ios-native/HughManatee/Models/WorkerModels.swift
import Foundation

// MARK: - Worker API Models (matching TypeScript worker.ts 1:1)

struct AgentConfigRequest: Codable {
    let first_name: String
    let birth_year: Int?
    let hometown: String?
    let voice_id: String
    let last_anchor: String?
    let preference: String?

    init(firstName: String, birthYear: Int?, hometown: String?, voiceId: String, lastAnchor: String?) {
        self.first_name = firstName
        self.birth_year = birthYear
        self.hometown = hometown
        self.voice_id = voiceId
        self.last_anchor = lastAnchor
        self.preference = nil
    }
}

struct AgentConfigResponse: Codable {
    let agent_id: String
    let conversation_token: String
    let first_turn: String
    let runtime_context: RuntimeContext
}

struct RuntimeContext: Codable {
    let seed_prompts: [String]
    let era_hooks: [String]
}

struct CollageRequest: Codable {
    let birth_year: Int?
    let hometown: String?
    let theme: String?
}

struct CollageResponse: Codable {
    let images: [CollageImage]
    let gradient: GradientColors
}

struct CollageImage: Codable {
    let url: String
    let alt: String
}

struct GradientColors: Codable {
    let from: String
    let to: String
}

struct AnchorRequest: Codable {
    let turns: [TurnPayload]
}

struct TurnPayload: Codable {
    let speaker: String
    let text: String
}

struct AnchorResponse: Codable {
    let anchor_phrase: String
    let title_suggestion: String
    let entities: [ExtractedEntity]
}

struct ExtractedEntity: Codable {
    let kind: String
    let value: String
}

struct WorkerError: Codable {
    let error: String
}

// MARK: - Voice Option

struct VoiceOption: Identifiable, Equatable {
    let id: String
    let voiceId: String
    let agentId: String
    let label: String
    let description: String

    init(voiceId: String, agentId: String, label: String, description: String) {
        self.id = voiceId
        self.voiceId = voiceId
        self.agentId = agentId
        self.label = label
        self.description = description
    }
}

let placeholderVoices: [VoiceOption] = [
    VoiceOption(voiceId: "voice-warm-female-01", agentId: "agent-warm-female-01",
                label: "Nora", description: "Warm, calm. A friend over tea."),
    VoiceOption(voiceId: "voice-warm-male-01", agentId: "agent-warm-male-01",
                label: "Arthur", description: "Steady, gentle. An older friend."),
    VoiceOption(voiceId: "voice-bright-female-01", agentId: "agent-bright-female-01",
                label: "June", description: "Brighter, curious. Asks good questions."),
]
